### PR TITLE
Catch validation exception and return error in callback

### PR DESCRIPTION
**Pull request by isocolsky**
This PR adds robustness to the interaction with the Insight server.

### CODE REVIEW

File: lib/insight.js [hunk 1]
```diff
@@ -64,7 +64,11 @@ Insight.prototype.getUnspentUtxos = function(addresses, callback) {
     if (err || res.statusCode !== 200) {
       return callback(err || res);
     }
-    unspent = _.map(unspent, UnspentOutput);
+    try {
+      unspent = _.map(unspent, UnspentOutput);
+    } catch (ex) {
+      return callback(ex);
+    }
```
Comment (maraoz) [inline lib/insight.js:73]: can you check for a specific error type here and throw the error if it isn't?

e.g:

```
catch (ex) {
  if (ex instanceof bitcore.errors) {
    return callback(ex);
  }
  throw ex;
}
```

Comment (isocolsky) [inline lib/insight.js:73]: yes, will do, thx

**Comment by isocolsky** [comment]
Ok, I'm catching the `InvalidArgument` error. It works but it is not perfect because of this code in `Bitcore.Transaction`:

``` javascript
  if (!txId || !JSUtil.isHexaString(txId) || txId.length > 64) {
    // TODO: Use the errors library
    throw new Error('Invalid TXID in object', data);
  }
  var outputIndex = _.isUndefined(data.vout) ? data.outputIndex : data.vout;
  if (!_.isNumber(outputIndex)) {
    throw new Error('Invalid outputIndex, received ' + outputIndex);
  }
```

**Comment by maraoz** [comment]
@isocolsky i'll open an issue to fix those error types. https://github.com/bitpay/bitcore/issues/1178

**Pull request merged**